docs/TESTCAFE.md

```diff
@@ -0,0 +1,49 @@
+## Cypress
+This guide aims to share how TestCafe is being used in Plio for performing E2E and integration tests.
+
+### Pre-requisites
+1. Run TestCafe for BrowserStack globally:
+    ```sh
+    npm i -g testcafe testcafe-browser-provider-browserstack
+    ```
+
+2. Add the following env variables to your `~/.bash_profile` or `~/.zshrc` file:
+    ```sh
+    export BROWSERSTACK_BASE_URL="http://localhost:8080"
+    export BROWSERSTACK_USERNAME="" # Your BrowserStack username
+    export BROWSERSTACK_ACCESS_KEY="" # Your BrowserStack access key
+    export BROWSERSTACK_PROJECT_NAME="plio"
+    ```
+
+3. Once the env variables are added, restart your terminal or run one of the following command:
+    ```sh
+    source ~/.bash_profile
+    # or
+    source ~/.zshrc
```

Comment:
So we need to install up `testcafe` globally on the user machine and not inside docker. Testcafe requires browsers installed so that it can open them and run the test cases within.

For example, here's the output of `testcafe --list-browsers` on my terminal:
![image](https://user-images.githubusercontent.com/12053186/150814337-1e073773-827e-493d-ada6-ce432be7bf18.png)

The output can be different for you depending on the browsers you've installed.

If I try to set up testcafe within the docker container, it cannot find the browsers installed and throws an empty list.
![image](https://user-images.githubusercontent.com/12053186/150816834-c503d28a-615b-4b4b-ad39-acba02e1204c.png)


Also, I did some research and testcafe has it's own docker image available. However it only comes with Chromium and Firefox.
https://testcafe.io/documentation/402838/guides/advanced-guides/use-testcafe-docker-image


I think let's keep it within user's machine for now. If needed, I can open up a task if it's possible to load local browsers from within docker. Another way would be to install browsers within our docker image but I believe we'll hit a road-block with Safari in that case. Let me know your thoughts.